### PR TITLE
[ShadowContainer][All Platforms] ShadowContainer not resizing properly

### PR DESCRIPTION
**Pull request by rafael-rosa-knowcode**
GitHub Issue (If applicable): closes: #798 

## PR Type

What kind of change does this PR introduce?

- Bugfix
- Refactoring (no functional changes, no api changes)

## What is the current behavior?
ShadowContainer is correctly stretching iwhen the content is changed to be larger but if the content is then shrunk, the ShadowContainer does not resize:

See Shadow Builder control inside of the Toolkit sample app:

shadowcontainerstretch
## What is the new behavior?

Removed some commented code and conditional layout refactoring.

Now the Canvas free up allocated space so content can reduce in size.

https://github.com/unoplatform/uno.toolkit.ui/assets/116665025/723a897e-40e8-43f3-b82d-ceb3cdbb99ec

## PR Checklist

Please check if your PR fulfills the following requirements:
- [ ] Tested code with current [supported SDKs](../README.md#supported)
- [ ] Tested the changes where applicable:
	- [ ] UWP
	- [x] WinUI
	- [ ] iOS
	- [ ] Android
	- [ ] WASM
	- [ ] MacOS
- [ ] Updated the documentation as needed:
	- [ ] [General Doc Update](https://github.com/unoplatform/uno.toolkit.ui/tree/main/doc)
	- [ ] [Controls Doc Update](https://github.com/unoplatform/uno.toolkit.ui/tree/main/doc/controls)
	- [ ] [Extensions Doc Update](https://github.com/unoplatform/uno.toolkit.ui/tree/main/doc/helpers)
	- [ ] [controls-styles.md](https://github.com/unoplatform/uno.toolkit.ui/blob/main/doc/controls-styles.md)
	- [ ] [lightweight-styling.md (LightWeight Styling Resource Keys)](https://github.com/unoplatform/uno.toolkit.ui/blob/main/doc/lightweight-styling.md)
- [ ] [Runtime Tests and/or UI Tests](https://platform.uno/docs/articles/contributing/guidelines/creating-tests.html) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Contains **NO** breaking changes
- [x] Associated with an issue (GitHub or internal)
- [x] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

### CODE REVIEW

**Review by agneszitte — COMMENTED** [review]
Can we add some tests for this @rafael-rosa-knowcode please?

**Comment by roubachof** [comment]
seems to be something weird happening with the positioning of the shadows when removing one (reducing size)

image

**Comment by rafael-rosa-knowcode** [comment]
Related to resizing, I updated the internal logic to fit the TextBox issue as the problem seemed similar.
However, the resizing situation in calculator still happens in android.

I isolated the ItemsControl and removed the shadows from it.
And it has the same behavior as Shadows.
Can you validate that it's a failure of the ItemsControl? Well, and I'm not finding a workaround.

Works on Skia. The isolated one and the shadows.


https://github.com/unoplatform/uno.toolkit.ui/assets/116665025/5e887c1f-6ded-4239-ba75-be6298fba3cd


But on Android does not.

https://github.com/unoplatform/uno.toolkit.ui/assets/116665025/51f2f25d-54e2-4ee6-941a-84610b612832

**Comment by rafael-rosa-knowcode** [comment]
@roubachof can you validate again?

File: src/Uno.Toolkit.Skia.WinUI/Controls/Shadows/ShadowContainer.cs [hunk 1]
```diff
@@ -256,9 +257,26 @@ protected override void OnApplyTemplate()
 	private void InvalidateCanvasLayout()
 	{
 		if (Content is not FrameworkElement contentAsFE ||
```
Comment (Xiaoy312) [inline src/Uno.Toolkit.Skia.WinUI/Controls/Shadows/ShadowContainer.cs:259]: dont align code, just indent them 1level per nesting
also even if you were to align them, it should be aligned to `if (`, not the `is not`

**Review by jeromelaban — APPROVED** [review]

**Review by Xiaoy312 — APPROVED** [review]

**Review by agneszitte — APPROVED** [review]

**Comment by agneszitte** [comment]
@Mergifyio backport legacy/3x release/stable/3.1

**Pull request merged**